fdbclient/FDBTypes.h

```diff
@@ -1441,16 +1442,34 @@ struct StorageMetadataType {
 	constexpr static FileIdentifier file_identifier = 732123;
 	// when the SS is initialized, in epoch seconds, comes from currentTime()
 	double createdTime;
+	KeyValueStoreType storeType;
+
+	// no need to serialize part (should be assigned after initialization)
+	bool wrongConfigured = false;
+
 	StorageMetadataType() : createdTime(0) {}
-	StorageMetadataType(uint64_t t) : createdTime(t) {}
+	StorageMetadataType(uint64_t t, KeyValueStoreType storeType = KeyValueStoreType::END, bool wrongConfigured = false)
+	  : createdTime(t), storeType(storeType), wrongConfigured(wrongConfigured) {}
 
 	static double currentTime() { return g_network->timer(); }
 
+	bool operator==(const StorageMetadataType& b) const {
+		return createdTime == b.createdTime && storeType == b.storeType && wrongConfigured && b.wrongConfigured;
+	}
+
+	bool operator<(const StorageMetadataType& b) const {
```

Comment:
The ordering of this class and how it is used is very hard to follow.

This `operator<` would sort `StorageMetadataType` objects in an order that is the reverse of the order that the PW should operate on them.  Then, the pair `MetadataUIDP` is defined plus a custom comparator that sorts by `StorageMetadataType` in _reverse_ order first via `a > b` and then by UID in natural order, but is commented as `less<T>` which is counter intuitive, and then this is used as the comparator for a max-heap which would have used `operator<` by default.

How about this instead:

- Define `operator<` here as sorting these objects in the order you would want them to be processed in for the PW
- Define `operator> (a, b)` here  as simply `return b < a` 
- Then without any further comparator definition,`std::pair<StorageMetadataType, UID>` would also sort in the order you want the UIDS to be processed
- You can then make a min-heap of your pair type, also without defining a new comparator, by using using `std::greater<MetadataUIDP>` as the comparator template argument, which is the usual way to turn a max-heap into a min-heap.
- Now your heap `top()` is the lowest element in the natural pair sort order which is the order the PW should operate on the items.